lms/lmsdb/models.py

```diff
@@ -132,21 +133,50 @@ def is_viewer(self) -> bool:
         return self.name == RoleOptions.VIEWER.value or self.is_manager
 
 
+class Course(BaseModel):
+    name = CharField(unique=True)
+    date = DateTimeField(default=datetime.now)
+    close_date = DateTimeField(null=True)
+    close_registration_date = DateTimeField(default=datetime.now)
+    invite_code = CharField(default=generate_invite_code, unique=True)
+    is_public = BooleanField(default=False)
+
+    def has_user(self, user_id: int) -> bool:
+        return UserCourse.is_user_registered(user_id, self)
+
+    @classmethod
+    def fetch(cls, user: 'User') -> Iterable['Course']:
+        return (
+            cls
+            .select()
+            .join(UserCourse)
+            .where(UserCourse.user == user.id)
+            .order_by(Course.name.desc())
+        )
+
+    def __str__(self):
+        return f'{self.name}: {self.date} - {self.close_date}'
+
+
 class User(UserMixin, BaseModel):
     username = CharField(unique=True)
     fullname = CharField()
     mail_address = CharField(unique=True)
     password = CharField()
     role = ForeignKeyField(Role, backref='users')
     api_key = CharField()
+    last_course_viewed = ForeignKeyField(Course, null=True)
     uuid = UUIDField(default=uuid4, unique=True)
 
     def get_id(self):
         return str(self.uuid)
 
-    def is_password_valid(self, password):
+    def is_password_valid(self, password) -> bool:
         return check_password_hash(self.password, password)
 
+    def is_registered(self, course_id: int) -> bool:
```

Comment:
Maybe `has_course`? Is registered without seeing the whole signature might be confusing